src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -482,7 +483,34 @@ void check(String apiUrl) throws IOException {
      * Otherwise this method throws {@link IOException} to indicate the problem.
      */
     public void checkApiUrlValidity() throws IOException {
-        retrieve().to("/", GHApiInfo.class).check(apiUrl);
+        try {
+            retrieve().to("/", GHApiInfo.class).check(apiUrl);
+        } catch (IOException ioe) {
+            if (isPrivateModeEnabled()) {
```

Comment:
Same feeling here, I would prefer a logic based on the HTTP error code (401 for private mode, http proxy auth...) and on the Network exception (connection refused, ssl handshake rejected for negotiation failure...). I am investigating to make proposals.